packages/frontend/src/contexts/state.js

```diff
@@ -1,15 +1,33 @@
 import { createContext } from 'react'
-import Info from './info'
 import Interface from './interface'
+import Info from './info'
+import Unirep from './unirep'
+import Attestation from './attestation'
+import StateTreeLeaf from './stateTreeLeaf'
+import Epoch from './epoch'
+import Attester from './attester'
+import User from './user'
 
 const state = {}
 
 const info = new Info(state)
 const ui = new Interface(state)
+const unirep = new Unirep(state)
+const attestation = new Attestation(state)
+const stateTreeLeaf = new StateTreeLeaf(state)
+const epoch = new Epoch(state)
+// const attester = new Attester(state)
+const user = new User(state)
 
 Object.assign(state, {
   info,
   ui,
+  unirep,
+  attestation,
+  stateTreeLeaf,
+  epoch,
+  // attester,
+  user,
 })
 
 export default createContext(state)
```

Comment:
re `state`, this comment also from UI PR:

> I recently changed this pattern so that we have a single state object that exports each store. This allows the stores to interact with each other without having to circularly import each other.

Did I understand correctly that we want to create the single state object containing the store from every route each time a page is rendered?

re `routes`:
I'm creating a route to grab data from each table in the db and just dumping it all onto `Home.jsx`. Probably not the most efficient approach(?) but I got data(!) and now I can see what I have access to and think about how to work with it. I'm planning to add/refine routes from here, please lmk if 
- what I've done so far makes sense
- you have opinions about what these routes should look like

(the attester route just isn't ready atm)

Reply:
See my other comment for more, but no, you don't want to load data in the `load` function, unless it's stuff the user will see immediately. You can avoid this too by having the component be responsible for loading the data. So if the user is loading an attester page, the component will ask the store to load data for the attester, then the store will populate the data.

The component should always render as if the data is there. So you might have a store `attesters` with an object `attestationsByAttesterId`. The component would do something like this:

```jsx
const AttesterComponent = ({ attesterId }) => {
  const { attesters } = state

  // This is an async function, but we're in a component, we don't care when it finishes
  attesters.loadAttestationsByAttesterId(attesterId)

  // This will initially render nothing, then when the call above ends it will render the
  // appropriate data
  return (
    <>
      ...stuff
      {attesters.attestationsByAttesterId[attesterId].map(({ id }) => <AttestationComponent attestationId={id} />)}
    </>
  )
}
```

This will render nothing by default, but as soon as data is loaded and added to the `attestationsByAttesterId` object, it will render the data the user wants to see.

You can add empty state or loading state handling like this:

```jsx
const AttesterComponent = ({ attesterId }) => {
  const { attesters } = state
  attesters.loadAttestationsByAttesterId(attesterId)
  return (
    <>
      ...stuff
      {attesters.attestationsByAttesterId[attesterId] ?
        attesters.attestationsByAttesterId[attesterId].map(({ id }) => <AttestationComponent attestationId={id} />) :
        null
      }
      {attesters.attestationsByAttesterId[attesterId] ?
        null :
        'Loading...!'
      }
    </>
  )
}
```